### PR TITLE
Migrate to single sln file + slnf files

### PR DESCRIPTION
**Pull request by Pilchie**

### CODE REVIEW

**Review by dougbu — APPROVED** [review]
This will be ready to go once you change enough `*.cs` files to get the remaining tests working. Some are likely looking for generated `*.sln` files but there are a lot of matches for `.sln` in `*.cs` files. Worse, a few like the match in src\Razor\Microsoft.AspNetCore.Razor.Language\test\Legacy\BaselineWriter.cs are exercised only when regenerating test baselines ☹️

**Comment by jkotalik** [comment]
Do we need to make an announcement for this (whether internal or external)? I wouldn't be surprised if some people's workflows may be disrupted by this change.

**Comment by Pilchie** [comment]
Good idea.  I can at least email the team internally.  Wonder if there are enough external contributors to make a post in aspnet/announcments, or maybe I just make a pinned issue for a bit?

**Comment by Tratcher** [comment]
Can you start by summarizing the impact on developers here?

**Comment by jkotalik** [comment]
Any of those options seem fine. Also could update the [getting started](https://github.com/dotnet/aspnetcore/blob/41ce223c1cf4ffd482c4ea36e88ed44a94db37b3/src/SignalR/docs/GettingStarted.md#getting-started).

**Comment by jkotalik** [comment]
I think there are some tests (server tests, blazor tests, IIS tests) that use the location of the sln to find specific files. I think you'll probably need to update them to slnf or update the test infrastructure to use a better path.

**Comment by pranavkm** [comment]
👍 to @jkotalik's suggestion. It'd be nice to point to the documentation on slnf files: https://docs.microsoft.com/en-us/visualstudio/ide/filtered-solutions?view=vs-2019 given slnf is a fairly new (and uncommonly seen-in-the-wild) feature.

**Comment by javiercn** [comment]
Not Blazor nor MVC tests. We do all that through assembly attributes we stamp on the test assembly

**Comment by javiercn** [comment]
> Good idea. I can at least email the team internally. Wonder if there are enough external contributors to make a post in aspnet/announcments, or maybe I just make a pinned issue for a bit?

Can we send an email to the team so that we get someone from each team to test their main area in case we missed something? 

Someone on @mkArtakMSFT team can go ahead and test MVC/Blazor and someone on the Core team can do the same thing for Servers/Signalr. /cc: @Tratcher @jkotalik before we merge.

I don't want this PR to be open for long since it'll get stale really quick.

**Comment by Pilchie** [comment]
I know @BrennanConroy took a look at the experience before.

**Comment by BrennanConroy** [comment]
> I know @BrennanConroy took a look at the experience before

Has anything changed?

**Comment by Pilchie** [comment]
Not really - just working on getting it updated for the blazor-wasm merge, and actually fixing tests this time.

File: src/ProjectTemplates/ProjectTemplatesNoDeps.slnf
Comment (pranavkm) [inline src/ProjectTemplates/ProjectTemplatesNoDeps.slnf:11]: You'll need to keep the `ComponentsWebAssembly-CSharp.sln` file that follows. It's part of the project template

Comment (Pilchie) [reply src/ProjectTemplates/ProjectTemplatesNoDeps.slnf:11]: Thanks, should be restored now.

**Comment by Pilchie** [comment]
@jkotalik @Tratcher @pranavkm would love it if you could take a look at the changes to `docs\BuildFromSource.md`, and see if you want more than that.

**Review by pranavkm — COMMENTED** [review]
BuildFromSource looks great.

**Comment by Tratcher** [comment]
The BuildFromSource description seems adequate, though I'm still wondering how many complaints we'll get about people trying to open the root sln directly.

The biggest downside I see to snlfs is the inability for a project to organize its view of the solution. The old slns usually had a Dependencies section to isolate the unrelated projects and allowed for a simpler structure for the relevant items.

**Comment by Tratcher** [comment]
> The BuildFromSource description seems adequate, though I'm still wondering how many complaints we'll get about people trying to open the root sln directly.

Can we rename it something less inviting like `SharedProjectDependencies.snl`, or move it to the eng folder?

**Comment by Pilchie** [comment]
Meh - I think VS is working on scaling better to solutions like this.  I was able to work on the full solution on my desktop for some tasks though the perf definitely wasn't what I'd like.

**Review by javiercn — APPROVED** [review]

**Comment by Pilchie** [comment]
If there are no other objections, I plan to merge this later this afternoon.

**Pull request merged**
